POSHIT/Example CHT application/malaria-usecase-cht/nools-extras.js

```diff
@@ -1,13 +1,72 @@
+const { DateTime } = require('luxon');
+const { FORMS } = require('./shared-extras');
+const { getField, isFirstReportNewer } = require('cht-nootils')();
+
 const isFormArraySubmittedInWindow = (reports, formsArray, startTime, endTime) => {
   if(typeof formsArray === 'string') { 
     formsArray = [ formsArray ];
   }
   return formsArray.some(f => Utils.isFormSubmittedInWindow(reports, f, startTime, endTime));
 };
+function ageInMonths(dateOfBirth) {
+  const dob = DateTime.fromISO(dateOfBirth);
+  const now = DateTime.now();
+  return now.diff(dob, 'months').months;
+}
 
-const isContactValid = (contact) => contact.contact && !contact.contact.date_of_death && !contact.contact.muted;
+// function getMostRecentReportForm(reports, form) {
+//   let result = null;
+//   reports.forEach(function(r) {
+//     if (form.indexOf(r.form) >= 0 &&
+//         !r.deleted &&
+//         (!result || r.reported_date > result.reported_date)) {
+//       result = r;
+//     }
+//   });
+//   return result;
+// }
+function getMostRecentReportForm(reports, form) {
+  let result;
+  reports.forEach(function (report) {
+    if (form.includes(report.form) &&
+      !report.deleted &&
+      (!result || report.reported_date > result.reported_date)) {
+      result = report;
+    }
+  });
+  return result;
+}
+const isAlive = (contact) => contact && contact.contact && !contact.contact.date_of_death;
+
+const isPregnancyTreatmentFollowUpForm = (report) => {
+  return FORMS.MALARIA_ASSESSMENT_FOR_PREGNANT_MOTHERS === report.form;
+};
+const isPregnancyTaskMuted = (contact) => {
+  const latestVisit = getMostRecentReportForm(contact.reports, [FORMS.MALARIA_ASSESSMENT_FOR_PREGNANT_MOTHERS]);
+  return latestVisit && isPregnancyTreatmentFollowUpForm(latestVisit) && getField(latestVisit, 'treatment_referral_follow_up_date') === '';
+};
+
+const isPregnancyBeforeDelivery = (report) => DateTime.now() < DateTime.fromISO(getField(report, 'group_pregnancy_registration_form.estimated_delivery_date'));
 
+const isFollowUpForm = (reports) => {
+  return isFirstReportNewer(getMostRecentReportForm(reports, FORMS.MALARIA_ASSESSMENT_FOR_PREGNANT_MOTHERS, getMostRecentReportForm(reports, FORMS.PREGNANCY_REGISTRATION)));
```

Comment:
This function also exists in cht-nootils

Reply:
@brian-gacheru , any reason for maintaining this?

Reply:
This function is not being used and has been removed